src/deepsparse/pipeline.py

```diff
@@ -142,13 +145,15 @@ def __init__(
         alias: Optional[str] = None,
         context: Optional[Context] = None,
         executor: Optional[Union[ThreadPoolExecutor, int]] = None,
+        logger: Optional[BaseLogger] = None,
     ):
         self._model_path_orig = model_path
         self._model_path = model_path
         self._engine_type = engine_type
         self._batch_size = batch_size
         self._alias = alias
         self.context = context
+        self.logger = logger
```

Comment:
We could add a `class NullLogger(BaseLogger)` to set self.logger to here so we don't have to wrap everything with `if self.logger`:

```python
self.logger = logger if logger is not None else NullLogger()
```

Reply:
Instead of adding another `Logger`, I'd rather just simply push the `if`-statement the `pipeline.log()` function. Same result, less coding.